source/extensions/common/wasm/wasm_vm.cc

```diff
@@ -72,11 +72,28 @@ bool EnvoyWasmVmIntegration::getNullVmFunction(std::string_view function_name, b
   return false;
 }
 
+bool isWasmEngineAvailable(absl::string_view runtime) {
+  auto runtime_factory = Registry::FactoryRegistry<WasmRuntimeFactory>::getFactory(runtime);
+  return runtime_factory != nullptr;
+}
+
+absl::string_view getFirstAvailableWasmEngineName() {
+  if (isWasmEngineAvailable("envoy.wasm.runtime.v8")) {
+    return "envoy.wasm.runtime.v8";
+  } else if (isWasmEngineAvailable("envoy.wasm.runtime.wasmtime")) {
+    return "envoy.wasm.runtime.wasmtime";
+  } else if (isWasmEngineAvailable("envoy.wasm.runtime.wamr")) {
+    return "envoy.wasm.runtime.wamr";
+  } else if (isWasmEngineAvailable("envoy.wasm.runtime.wavm")) {
+    return "envoy.wasm.runtime.wavm";
+  }
+  return "";
+}
+
 WasmVmPtr createWasmVm(absl::string_view runtime) {
+  // Set wasm runtime to build-in wasm engine if it is not specified
```

Comment:
Nit: `build-in` -> `built-in` and `wasm` -> `Wasm`.

Reply:
Fixed.